gbpservice/nfp/configurator/agents/agent_base.py

```diff
@@ -0,0 +1,181 @@
+from oslo_log import log as logging
+from gbpservice.nfp.configurator.lib import (
+    generic_config_constants as const)
+
+LOG = logging.getLogger(__name__)
+
+"""Implements base class for all service agents.
+
+Common methods for service agents are implemented in this class. Configurator
+module invokes these methods through the service agent's child class instance.
+
+"""
+
+
+class AgentBaseRPCManager(object):
+
+    def __init__(self, sc, conf):
+        self._sc = sc
+        self.conf = conf
+
+    def validate_request(self, sa_info_list, notification_data):
+        if (isinstance(sa_info_list, list) and
+                isinstance(notification_data, dict)):
+            return True
+        else:
+            return False
+
+    def forward_request(self, sa_info_list, notification_data):
```

Comment:
Actual request is processed by the service agents. The base agent is forwarding the request, so is the name. We can discuss on this.
sa_req_list name makes more sense than sa_info_list. Will change it.